tempesta_fw/sock_clnt.c

```diff
@@ -123,15 +121,21 @@ tfw_sock_clnt_new(struct sock *sk)
 		goto err_conn_init;
 	}
 
+	spin_lock(&conn->splock);
+	tfw_connection_link_sk(conn, sk);
+	tfw_connection_link_peer(conn, (TfwPeer *)cli);
```

Comment:
TfwPeer lives longer than TfwConnection by its definition, so access to it should not be managed by shorter living objects like TfwConnection. Moreover, the unlink operation is synchronized by TfwPeer->conn_lock.

It's unclear why we need to synchronize access to conn->peer. If it must be synchronized, then probably we should do this in different way...